transport/src/main/java/io/netty/bootstrap/AbstractBootstrap.java

```diff
@@ -218,6 +220,12 @@ public B validate() {
         return (B) this;
     }
 
+    private void ensureChannelFactory() {
+        if (channelFactory == null) {
+            channelFactory = ChannelFactoriesRegistry.getFactoryForEventLoopGroup(group.getClass(), isServerSide());
```

Comment:
Nah, should behave the same as right now saying something along the lines of ChannelFactory being missing.

You wouldn't want to return a client channel factory by default if the user's custom bootstrap is for server channels.